docs/book/src/tasks/experimental-features/cluster-class/operate-cluster.md

```diff
@@ -266,6 +267,76 @@ Please note that:
 
 </aside>
 
+# Upgrading Cluster API
+
+There are some special considerations for ClusterClass regarding Cluster API upgrades when the upgrade includes bumping
+the apiVersion of infrastructure, bootstrap or control plane provider CRDs.
+
+In general the Cluster topology controller always uses exactly the versions of the CRDs referenced in the ClusterClass.
+This means in the following example the Cluster topology controller uses `v1beta1` even if the `DockerClusterTemplate` already 
+has a `v1beta2` version.
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: quick-start
+  namespace: default
+spec:
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: DockerClusterTemplate
+...
+```
+
+It is recommended to first upgrade Cluster API and then update the apiVersions in the ClusterClass afterwards. There won't be 
+any disruptions of the reconciliation as the Cluster topology controller is able to reconcile the Cluster even with the old 
+apiVersion in the ClusterClass. 
```

Comment:
what about moving this up as a second paragraph, given that the example above is the explanation of why there won't be disruption

Reply:
also, let's add a note about the fact that when upgrading apiVersion for references, corresponding patches should be checked and eventually changed according to API changes; in the case of external patches, before completing this operation it is required to ensure that all the external patches support the new apiVersion.
an example of how references and patches should be upgraded could really help

Reply:
I saw this is already documented below, but we could benefit from a better separation from instruction for users from instruction for the external patches authors.

Reply:
> what about moving this up as a second paragraph, given that the example above is the explanation of why there won't be disruption

Done

Reply:
I gave it a try :)